rllib/connectors/tests/test_agent.py

```diff
@@ -196,39 +194,30 @@ def test_vr_connector_respects_training_or_inference_vr_flags(self):
             is_policy_recurrent=True,
         )
 
-        for_action_expected = SampleBatch(
+        sample_batch_expected = SampleBatch(
             {
                 "both": obs_arr[None],
                 "only_inference": obs_arr[None],
                 "seq_lens": np.array([1]),
             }
         )
 
-        for_training_expected_list = [
-            # is_training = False
-            None,
-            # is_training = True
-            agent_data,
-        ]
-
-        for is_training in [True, False]:
-            c = ViewRequirementAgentConnector(ctx)
-            c.is_training(is_training)
-            processed = c([data])
+        c = ViewRequirementAgentConnector(ctx)
+        c.in_training()
```

Comment:
If we call c.in_training(), why do we expect the `only_inference` field to be present instead of the `only_training ` field in `sample_batch_expected`?

Reply:
ah, the story is that I realized we need to provide all columns for training, even if the column itself has used_for_training=False.
because for example, info dicts may contain important metadata for their model, or a new column may need data from another column that itself isn't actually used_for_training.
long story short is that before we move the entire trajectory construction into connectors, we need to output the raw dict here for SampleCollector.

Reply:
Ok ok got it, that's the point of the whole PR I guess 😃 Just found it counterintuitive that the "used_for_training" field would not make it through. Thanks!

Reply:
yeah, definitely. super mind-twisting. as soon as we have the full trajectory in connectors, we can handle all the view_requirements transformation in there.
then we may not need to output anything not for training. if someone needs infos dict for their model, they need to mark it as so.